asmrun/signals_asm.c

```diff
@@ -214,38 +214,69 @@ DECLARE_SIGNAL_HANDLER(segv_handler)
   struct rlimit limit;
   struct sigaction act;
   char * fault_addr;
+  int handled = 0;
```

Comment:
I'm no fan of Boolean variables that encode control flow.  Do you really need this variable or would the if...else... do?